### PR TITLE
feat(extension): OnBoarding

### PR DESCRIPTION
**Pull request by gregoirelacoste**
linked to #837

### CODE REVIEW

**Review by christpet — APPROVED** [review]
Looks good to me... @gregoirelacoste is this just to test?

**Comment by newick** [comment]
@gregoirelacoste I put it in draft ;)

**Comment by gregoirelacoste** [comment]
@newick Thx, @christpet sorry, I just created this to have a referral PR during the code

**Comment by gregoirelacoste** [comment]
@newick If you want to work from storybook for the rest, you can now, components are ready

**Comment by gregoirelacoste** [comment]
@christpet @lutangar @newick almost finish my part.
Details on how onboarding will appear: we will display the OnBoarding component only on the /eclaireurs page and only when the user come from the first time from the installation extension page. 
After install extension, users are redirected to /eclaireurs?pk_campaign=installed.
I would like to use react-router hook to get params, but it seems that our version (v4) haven't get yet, so i did use window.location.search. Looks good to you ?

Is there a Grid system on the app ? I didn't find one, so I tried to DIY one

**Comment by lutangar** [comment]
> __Note:__
> @dis-moi/devs prefer the `draft` __Github__ mechanism over a `wip` label for PRs which aren't ready to be merged. 
> ( as @newick did here)
> https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Comment by lutangar** [comment]
> I would like to use react-router hook to get params

Dunno about hooks, but if we're speaking about query parameters, they should always be accessible somewhere in the props.
The router has to be injected in some way. 
Have a look at `selectors` using `makeGetRouteParam`, it might help in solving your issue, and look where and how they are used.

About the grid I don't know but @newick would. I'd follow his recommendations on this.

> almost finish my part.

@gregoirelacoste let me know when it's done and I'll review this.

**Review by gregoirelacoste — COMMENTED** [review]
Nice!

**Comment by newick** [comment]
@gregoirelacoste could you check if a `SetTimeOut` could improve the feeling at the end of the video?

**Comment by gregoirelacoste** [comment]
cc https://github.com/dis-moi/extension/issues/837#issuecomment-805741590

**Review by christpet — APPROVED** [review]
Looks good, merci @gregoirelacoste !

**Comment by christpet** [comment]
@gregoirelacoste looks like it's still a draft, how can we merge this?

**Comment by gregoirelacoste** [comment]
THX @christpet ! it's not over yet, there may be a new version of the video? cc  @newick  and it would be nice to have a @lutangar + @newick  review ?

**Comment by christpet** [comment]
Let's push it to staging with the existing video. I will let @newick and @lutangar weigh in before that though.

**Comment by gregoirelacoste** [comment]
I'm afraid the last step is not yet perfect
image

**Comment by newick** [comment]
Step 3 has been sent to Marco for design.

**Comment by newick** [comment]
@gregoirelacoste here is my shopping list :) :
* `<Text>Oui, j&apos; épinglé DisMoi</Text>` in Step2 needs a "ai" after the `&apos;`
* an `overflow: hidden` on `body` would be awesome when the modale is opened to hide the scrollbar
* the loader bar on Step2 starts a long time after Bullito, didn't find where it comes from

Thanks a lot!

File: src/components/atoms/Grid/index.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  justify-content: ${props => props.justifyContent || 'center'};
+  gap: ${props => props.gap + 'em' || 0};
+`;
+
+const GridItem = styled.div<GridProps>`
+  flex: 1;
+  flex-basis: 100%;
+`;
+
+export { GridContainer, GridItem };
```
Comment (newick) [inline src/components/atoms/Grid/index.tsx:23]: we can remove these files, they are no use so far

**Review by newick — CHANGES_REQUESTED** [review]

File: src/components/organisms/OnBoarding/components/Buttons.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import styled from 'styled-components';
+import { BackgroundButton } from 'components/atoms';
+
+type Color = string | undefined;
+const BackgroundColor = (color: Color) => {
+  switch (color) {
```
Comment (lutangar) [inline src/components/organisms/OnBoarding/components/Buttons.tsx:6]: Switch for only 2 cases?

Comment (lutangar) [reply src/components/organisms/OnBoarding/components/Buttons.tsx:6]: Intuitively a UpperCamelCase variable is a Class/Interface and here it's a function.
Function name could be smthing like `getBackgroundColor`.
A `Color` that can be the string `inactive` is not very intuitive.

File: src/components/organisms/OnBoarding/components/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import styled from 'styled-components';
+import CloseButton from '../../Notification/NotificationHeader/CloseButton';
+import { PopinClose } from 'components/molecules/Popin/Popin';
+import { CloseFunction } from '../index';
```
Comment (lutangar) [inline src/components/organisms/OnBoarding/components/Modal.tsx:5]: `../`

File: src/components/organisms/OnBoarding/components/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import useNoScrollBody from '../../../../app/utils/useNoScrollBody';
```
Comment (lutangar) [inline src/components/organisms/OnBoarding/components/Modal.tsx:6]: Lot's of `..`.
I suppose we could use an `effects` now that we have a few.

**Review by lutangar — CHANGES_REQUESTED** [review]
Overall:
- `src/components/organisms/OnBoarding/components/` I don't get the point of the extra components dir. we already are in a components folder.
- translations?

File: src/components/atoms/Grid/index.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import styled from 'styled-components';
+
+interface GridProps {
+  direction?: 'column';
+  justifyContent?: 'center' | 'space-around' | 'flex-start';
+  gap?: number;
+}
+
+const GridContainer = styled.div<GridProps>`
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/components/atoms/Grid/index.tsx:9]: just deleted this components, we don't use it finally

File: src/components/atoms/Paragraph/Paragraph.ts [hunk 6]
```diff
@@ -1,9 +1,14 @@
...
-const Paragraph = styled.p`
+interface ParagraphProps {
+  align?: 'right' | 'left' | 'center';
+}
+
+const Paragraph = styled.p<ParagraphProps>`
   margin: 0;
   font-size: ${props => props.theme.fontSizeDefault};
   color: ${props => props.theme.textColor};
+  text-align: ${props => props.align || 'center'};
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/components/atoms/Paragraph/Paragraph.ts:11]: indeed

File: src/components/atoms/Title/Title.ts [hunk 7]
```diff
@@ -1,8 +1,12 @@
 import styled from 'styled-components';
 
-export default styled.h2`
+interface H2Props {
+  align?: 'right' | 'left' | 'center';
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/components/atoms/Title/Title.ts:4]: that's weird, it said nothing on my PC...

File: src/components/atoms/Title1/Title1.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components';
+
+export default styled.h1`
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/components/atoms/Title1/Title1.ts:3]: Not relevant for our use

File: src/app/profiles/App/Pages/Profiles/List/ProfileList.tsx [hunk 9]
```diff
@@ -127,6 +129,7 @@ const ProfileList = ({
 
   return (
     <>
+      {isRedirectedFromExtensionInstall && <OnBoarding />}
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/app/profiles/App/Pages/Profiles/List/ProfileList.tsx:132]: prop of which component ? Profile list on onboarding ?
I can also inject isRedirectedFromExtensionInstall directly into OnBoarding with something lik : 
`if(!isRedirectedFromExtensionInstall) return null` ?

File: src/components/organisms/OnBoarding/Steps/Step3.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+  color: ${props => props.theme.primaryColor};
+
+  &:first-of-type {
+    margin-top: 60px;
+  }
+`;
+
+export default ({ prev, close }: StepTypes) => (
+  <Content>
+    <MarginTitle>Bienvenue chez DisMoi</MarginTitle>
```
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/components/organisms/OnBoarding/Steps/Step3.tsx:26]: @christpet ? could you provide a translation ?

File: src/components/organisms/OnBoarding/components/Buttons.tsx (hunk 2 above)
Comment (newick) [reply src/components/organisms/OnBoarding/components/Buttons.tsx:6]: My bad, the idea is to clean all the Buttons and have some props for everything (color, size, etc.)

What's the best way until then?

File: src/components/organisms/OnBoarding/components/Modal.tsx (hunk 4 above)
Comment (gregoirelacoste) [reply src/components/organisms/OnBoarding/components/Modal.tsx:6]: i didn't get it, sorry ?

**Comment by newick** [comment]
>      `src/components/organisms/OnBoarding/components/` I don't get the point of the extra components dir. we already are in a components folder.

Maybe @gregoirelacoste let's leave the components section and have an onboarding one, I'll be happy with that!

**Comment by gregoirelacoste** [comment]
@newick good idea, what do you think about that @lutangar ?

File: src/components/organisms/OnBoarding/Steps/Step3.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (gregoirelacoste) [reply src/components/organisms/OnBoarding/Steps/Step3.tsx:26]: translation function implementation check

**Comment by gregoirelacoste** [comment]
@christpet we need translation, i just implement the function and updated the translation file

File: src/app/profiles/App/Pages/Profiles/List/ProfileList.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (lutangar) [reply src/app/profiles/App/Pages/Profiles/List/ProfileList.tsx]: > I can also inject isRedirectedFromExtensionInstall directly into OnBoarding

Sounds good. The only important thing is it should be a prop. It'll make your life easier on storybook as well.

File: src/components/organisms/OnBoarding/Steps/Step3.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (lutangar) [reply src/components/organisms/OnBoarding/Steps/Step3.tsx]: > translation function implementation check 

:question:

File: src/components/organisms/OnBoarding/components/Buttons.tsx (hunk 2 above)
Comment (lutangar) [reply src/components/organisms/OnBoarding/components/Buttons.tsx]: Until then just list specific strings like `inactive` (like an enum) in your type :
```
type Color = 'inactive' | string | undefined;
```
Is color the best name for these ?

You could prepare the ground and do smthing like this:
```
type ButtonState = 'inactive' | undefined;
type Color = string | ButtonState;
```

File: src/components/organisms/OnBoarding/components/Modal.tsx (hunk 4 above)
Comment (lutangar) [reply src/components/organisms/OnBoarding/components/Modal.tsx]: Put all effects `useNoScrollBody` `useThis` `useThat` into an effect `directory`

Comment (lutangar) [reply src/components/organisms/OnBoarding/components/Modal.tsx]: I mean if there are more than 1

Comment (gregoirelacoste) [reply src/components/organisms/OnBoarding/components/Modal.tsx]: ah okay

File: src/components/organisms/OnBoarding/components/Buttons.tsx (hunk 2 above)
Comment (newick) [reply src/components/organisms/OnBoarding/components/Buttons.tsx]: I think you lost me. 

How to use that?

Comment (lutangar) [reply src/components/organisms/OnBoarding/components/Buttons.tsx]: `Ctrl+V` then `Ctrl+V` to replace `type Color = string | undefined;`

Comment (newick) [reply src/components/organisms/OnBoarding/components/Buttons.tsx]: and you mean that I keep the switch?

I don't get the point but m'kay ^^

Comment (lutangar) [reply src/components/organisms/OnBoarding/components/Buttons.tsx]: I feel like I'm stealing a kid toy... :car: 
"You may keep the `switch` this time. But next time you have only 2 cases, you must promesse me you'll use an `if ... else`."

Comment (gregoirelacoste) [reply src/components/organisms/OnBoarding/components/Buttons.tsx]: @lutangar I changed that yesterday to put a if statement, you didn't see my changes ?

Comment (newick) [reply src/components/organisms/OnBoarding/components/Buttons.tsx]: > here @lutangar @newick
> reassure me, you also see the translations?

I see the translations, yep.
Oh and yeah there's a `if`, didn't checked it, my bad 🤦

**Review by lutangar — APPROVED** [review]

**Comment by lutangar** [comment]
Rebase and you should be good to go!

**Review by newick — APPROVED** [review]

**Comment by gregoirelacoste** [comment]
@christpet you can translate now, it's clean !

File: src/i18n/resources/en/profiles.json [hunk 11]
```diff
@@ -18,12 +18,14 @@
...
-    "filter_by": "Filter by:"
+    "filter_by": "Filter by:",
+    "finish": "End"
   },
   "error": {
     "common": {
       "title": "Oops!",
-      "message": "Something went wrong. Please restart the extension or try again later."
+      "message": "Something went wrong. Please restart the extension or try again later.",
+      "browser_old": "Votre navigateur ne gère pas l'élément"
```
Comment (christpet) [inline src/i18n/resources/en/profiles.json:28]:
```suggestion
      "browser_old": "Your browser can't handle this element. Please update your browser to the latest version."
```

File: src/i18n/resources/en/profiles.json [hunk 12]
```diff
@@ -67,6 +69,23 @@
...
+    },
+    "onBoarding": {
+      "step1": "L'installation a réussi !<br/>  Chargement de votre expérience...<br/>  Épinglez DisMoi",
+      "step2": {
+        "title": "Pin DisMoi",
+        "question": "Tu es arrivé à épingler l'extension DisMoi ?",
+        "answer1": "Understood",
+        "button1": "Revoir le tuto",
+        "answer2": "Oui, j'ai épinglé DisMoi",
+        "button2": "Dernière étape"
```
Comment (christpet) [inline src/i18n/resources/en/profiles.json:81]:
```suggestion
      "step1": "DisMoi is installed!<br/>  Your experience is loading...<br/>  Pin DisMoi in your Browser",
      "step2": {
        "title": "Pin DisMoi",
        "question": "Did you pin the DisMoi extension in your browser?",
        "answer1": "Understood",
        "button1": "View the tutorial again",
        "answer2": "Yes, I pinned DisMoi",
        "button2": "Final step"
```

File: src/i18n/resources/en/profiles.json [hunk 13]
```diff
@@ -67,6 +69,23 @@
...
+      },
+      "step3": {
+        "title": "Bienvenue chez DisMoi",
+        "sub_title":"Un dernier mot !",
+        "paragraph": "<0>- Lors de l'installation, nous vous avons présélectionné plusieurs éclaireurs, plutôt consensuels. À vous de compléter et faire les choix définitifs !</0><1>- Une fois que vous avez terminé, vous pouvez revenir à votre navigation normale sur le web. Les conseils et infos de vos éclaireurs apparaitront directement sur les pages web que vous visitez.</1>",
+        "back_button": "Revoir le tuto"
```
Comment (christpet) [inline src/i18n/resources/en/profiles.json:87]:
```suggestion
        "title": "Welcome to DisMoi",
        "sub_title":"One final word!",
        "paragraph": "<0>- We have preselected several Guides for you to follow. Now you can decide who to follow!</0><1>- Once you're done, you can go back to browsing the web normally. Posts from Guides you follow will appear directly on the web pages you visit.</1>",
        "back_button": "View the tutorial again"
```

**Review by christpet — APPROVED** [review]
This is good!

**Comment by christpet** [comment]
@gregoirelacoste OK I think the translation is complete - tell me if I'm wrong!

**Comment by gregoirelacoste** [comment]
@christpet LGTM ! i merge to master if ok for you

**Comment by christpet** [comment]
@gregoirelacoste just tested this on Brave and Chrome, result is that it looks like the tutorial skips a step and stays stuck on step 3: 

https://user-images.githubusercontent.com/32476396/112841313-a0179800-90a0-11eb-9687-5a682d40ed5d.mov

**Pull request merged**